enumerables.rb

```diff
@@ -0,0 +1,134 @@
+module Enumerable
+  # My Each Method
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self.class == Array ? self : to_a
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter])
+      counter += 1
+    end
+    arr
+  end
+
+  # My Each With Index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self.class == Array ? self : to_a
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter], counter)
+      counter += 1
+    end
+    arr
+  end
+
+  # My Select
+  def my_select(arr)
+    new_array = []
+    arr.my_each do |index_value|
+      new_array << index_value if yield(index_value)
+    end
+    new_array
+  end
+
+  # My All?
+  def my_all?(arr)
+    result = false
+    arr.my_each do |item|
+      result = true if yield(item == true)
+    end
+    result
+  end
+
+  # My Any?
+  def my_any?(arr)
+    result = false
+    arr.my_each do |item|
+      next unless yield(item)
+
+      result = true
+    end
+    result
+  end
+
+  # My None?
+  def my_none?(arr)
+    result = true
+    arr.my_each do |item|
+      result = false if yield(item)
+    end
+    result
+  end
+
+  # My Count
+  def my_count(arr)
+    counter = 0
+    arr.my_each do |item|
+      counter += 1 if yield(item)
+    end
+    counter
+  end
+
+  # My Map
+  def my_map(arr, &block_given)
+    result = []
+    arr.my_each do |element|
+      result << block_given.call(element)
+    end
+    result
+  end
+
+  # My Inject
+  def my_inject(arr, start_value = 0)
+    counter = 0
+    accumulator = start_value
+    while counter < arr.size
+      accumulator = yield(accumulator, arr[counter])
+      counter += 1
+    end
+    accumulator
+  end
+
+  # Multiply LS
+  def multiply_els(arr)
+    arr.my_inject(arr, 1) { |a, b| a * b }
+  end
+end
```

Comment:
Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)

Reply:
Fixed and tested